sklearn/ensemble/tests/test_gradient_boosting_loss_functions.py

```diff
@@ -184,3 +185,30 @@ def test_sample_weight_deviance():
         deviance_w_w = loss(y, p, sample_weight)
         deviance_wo_w = loss(y, p)
         assert deviance_wo_w == deviance_w_w
+
+
+def test_multinomial_deviance():
+    # Check multinomial deviance with and without sample weights.
+    rng = check_random_state(13)
+    sample_weight = np.ones(100)
+    y = rng.randint(0, 3, size=sample_weight.shape[0])
+    p = np.zeros((y.shape[0], 3), dtype=np.float64)
+    for i in range(p.shape[1]):
+        p[:, i] = y == i
+
+    assert_raises(ValueError, MultinomialDeviance, 2)
+    loss = MultinomialDeviance(3)
+    loss_wo_sw = loss(y, p)
+    assert loss_wo_sw > 0
+    loss_w_sw = loss(y, p, np.ones(p.shape[0], dtype=np.float32))
+    assert_almost_equal(loss_wo_sw, loss_w_sw)
+    loss_w_sw = loss(y, p, 0.5 * np.ones(p.shape[0], dtype=np.float32))
+    assert_almost_equal(loss_wo_sw, loss_w_sw)
+
+    # second check
```

Comment:
I would make this its own test

```py
def test_loss_computation(pred=np.array([[1.0, 0, 0],
￼                                         [0, 0.5, 0.5]]),
                          y=np.array([0, 1]),
                          weights=np.array([1, 3]),
                          expected_loss=0.85637):
   assert_almost_equal(loss(y, pred, weights), expected_loss, decimal=4)
```

Maybe even add a test for `weights=None`.

Reply:
I would also split the rest of the test. I think that testing that the error raising should go on its own, etc..